### PR TITLE
Add org.jellyfin.JellyfinServer

### PR DESCRIPTION
**Pull request by istori1**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.jellyfin.Jellyfin.yml [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+
+cleanup:
+  # - /include
+  # - /lib/cmake
+  # - /lib/pkgconfig
+  - /man
+  - /share/man
+
+finish-args:
+  - --device=dri
```
Comment (hfiguiere) [inline org.jellyfin.Jellyfin.yml:20]: so it displays nothing?

File: org.jellyfin.Jellyfin.yml [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+      - --extra-libs=-lfftw3f
+    sources:
+      - type: archive
+        url: https://github.com/jellyfin/jellyfin-ffmpeg/archive/refs/tags/v5.1.2-8.tar.gz
+        sha256: 51796d269389c42fff4e0cb4efd4c79a6e533def9e62caa0a8bb3965bbef92a7
+      - type: shell
+        commands:
+          - for n in $(cat debian/patches/series); do if [[ $n != "#"* ]]; then patch -Np1 -i "debian/patches/$n" -d .; fi; done
+    modules:
+      - name: fftw3
```
Comment (hfiguiere) [inline org.jellyfin.Jellyfin.yml:98]: please use the shared-module.

File: org.jellyfin.Jellyfin.yml [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+        buildsystem: cmake
+        config-opts:
+          - -DCMAKE_BUILD_TYPE=Release
+          - -DBUILD_SHARED_LIBS=ON
+          - -DBUILD_TOOLS=OFF
+          - -DBUILD_TESTS=OFF
+          - -DFFT_LIB=fftw3f
+        sources:
+          - type: archive
+            url: http://archive.ubuntu.com/ubuntu/pool/universe/c/chromaprint/chromaprint_1.5.1.orig.tar.gz
```
Comment (hfiguiere) [inline org.jellyfin.Jellyfin.yml:141]: please avoid using ubuntu archives.

File: org.jellyfin.Jellyfin.yml [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+            url: http://archive.ubuntu.com/ubuntu/pool/main/f/fftw3/fftw3_3.3.8.orig.tar.gz
+            sha256: 6113262f6e92c5bd474f2875fa1b01054c4ad5040f6b0da7c03c98821d9ae303
+          - type: archive
+            url: http://archive.ubuntu.com/ubuntu/pool/main/f/fftw3/fftw3_3.3.8-2ubuntu8.debian.tar.xz
+            sha256: 2bddf40b42383167b221d7adfc7937b102b366053c3e6055f3864fb7b98c3050
+          - type: shell
+            commands:
+              - for n in $(cat patches/series); do if [[ $n != "#"* ]]; then patch -Np1 -i "patches/$n" -d .; fi; done
+      - name: chromaprint
+        buildsystem: cmake
```
Comment (hfiguiere) [inline org.jellyfin.Jellyfin.yml:132]:
```suggestion
        buildsystem: cmake-ninja
```

File: org.jellyfin.Jellyfin.yml (hunk 3 above)
Comment (hfiguiere) [reply org.jellyfin.Jellyfin.yml:141]: this apply to the rest.

File: org.jellyfin.Jellyfin.yml [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+            url: http://archive.ubuntu.com/ubuntu/pool/universe/z/zvbi/zvbi_0.2.41.orig.tar.gz
+            sha256: 2f22f9846bceead26349dd0b026cd41b638171eca2af98e69fe6762b98cbcd87
+          - type: archive
+            url: http://archive.ubuntu.com/ubuntu/pool/universe/z/zvbi/zvbi_0.2.41-1.debian.tar.xz
+            sha256: 4a8767a1fb6fcfdcf9da21945a33de9c9cfac3ebfa4222929a043fabc1110492
+          - type: shell
+            commands:
+              - for n in $(cat patches/series); do if [[ $n != "#"* ]]; then patch -Np1 -i "patches/$n" -d .; fi; done
+      - name: khronos-opencl-headers
+        buildsystem: cmake
```
Comment (hfiguiere) [inline org.jellyfin.Jellyfin.yml:327]:
```suggestion
        buildsystem: cmake-ninja
```

File: org.jellyfin.Jellyfin.yml [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+          - mkdir -p $FLATPAK_DEST/include
+          - mv amf/public/include $FLATPAK_DEST/include/AMF
+        cleanup:
+          - "*"
+        sources:
+          - type: archive
+            url: https://github.com/GPUOpen-LibrariesAndSDKs/AMF/archive/refs/tags/v1.4.28.tar.gz
+            sha256: 22524c74ca47686a61e9d385b66bbb4735b680274f7c7d8ca3c3b6d2877ec00a
+      - name: svt-av1
+        buildsystem: cmake
```
Comment (hfiguiere) [inline org.jellyfin.Jellyfin.yml:366]:
```suggestion
        buildsystem: cmake-ninja
```

File: org.jellyfin.Jellyfin.yml [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+      - --enable-version3
+      - --enable-vulkan
+      - --extra-libs=-lfftw3f
+    sources:
+      - type: archive
+        url: https://github.com/jellyfin/jellyfin-ffmpeg/archive/refs/tags/v5.1.2-9.tar.gz
+        sha256: 5eefa589d1fe7b8878f7fa4dd71a61b5902299e420cc9e0ef1a406d288feeeee
+      - type: shell
+        commands:
+          - for n in $(cat debian/patches/series); do if [[ $n != "#"* ]]; then patch -Np1 -i "debian/patches/$n" -d .; fi; done
```
Comment (hfiguiere) [inline org.jellyfin.Jellyfin.yml:96]: I doubt this work.

Comment (istori1) [reply org.jellyfin.Jellyfin.yml]: These patches are included in the repo.

File: org.jellyfin.JellyfinServer.yml [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+          - --disable-examples
+          - --disable-bdjava-jar
+        sources:
+          - type: archive
+            url: https://download.videolan.org/pub/videolan/libbluray/1.3.2/libbluray-1.3.2.tar.bz2
+            sha256: 456814db9f07c1eecdef7e840fcbb20976ef814df875428bfb81ecf45851f170
+      - name: intel-mediasdk
+        only-arches:
+          - x86_64
+        buildsystem: cmake
```
Comment (hfiguiere) [inline org.jellyfin.JellyfinServer.yml:139]: cmake-ninja doesn't work?

**Comment by hfiguiere** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by hfiguiere** [comment]
appstream file can't be found.

**Comment by hfiguiere** [comment]
bot, build org.jellyfin.JellyfinServer

File: org.jellyfin.JellyfinServer.yml (hunk 8 above)
Comment (istori1) [reply org.jellyfin.JellyfinServer.yml]: I missed this one. It is now cmake-ninja

**Comment by istori1** [comment]
> Build [30548 failed](https://buildbot.flathub.org/#/builders/6/builds/30548)

@hfiguiere Looks like wrong casing made it fail. I added a new commit fixing the case. Could there be something else?

**Comment by hfiguiere** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by barthalion** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by barthalion** [comment]
bot, build org.jellyfin.JellyfinServer

File: jellyfin-launcher.sh [hunk 9]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+PORT=8096
+
+if ! curl --silent http://localhost:$PORT > /dev/null; then
+  flatpak run org.jellyfin.Jellyfin & sleep 10
```
Comment (hfiguiere) [inline jellyfin-launcher.sh:6]: Since this script is the `command`, this is completely wrong. You don't `flatpak run` inside flatpak.

**Comment by hfiguiere** [comment]
```
Failed to download sources: module svt-av1: Wrong sha256 checksum for SVT-AV1-v1.2.1.tar.gz, expected "5c2466eb2990fa045b29c63b92a2b5f13d9cb841709df5e7206f96a59e119fdd", was "6834e7e5f4c1f9e5012d3fd4a29b770944c4ba95cde110b3bd29b33e385ee08c"
```

File: jellyfin-launcher.sh (hunk 9 above)
Comment (istori1) [reply jellyfin-launcher.sh]: Updated script

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by barthalion** [comment]
>  I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub

Could you elaborate on your connection to the upstream team, or where the discussion has taken place?

**Comment by istori1** [comment]
> > I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub
> 
> Could you elaborate on your connection to the upstream team, or where the discussion has taken place?

The initial discusion was around two months ago on a public matrix room. They where not interested in building a flatpak, but welcomed the idea of having it created for them as a comunity supported project. I am also an active participant on their developmet rooms.

**Comment by orowith2os** [comment]
> They where not interested in building a flatpak, but welcomed the idea of having it created for them as a comunity supported project. I am also an active participant on their developmet rooms.

Have you done any testing to ensure this works properly compared to, say, a docker container? It could also be useful to upstream some of these files, like the .desktop and .metainfo.xml files.

File: org.jellyfin.JellyfinServer.yml [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+  - --share=network
+  # - --device=all # Needed for v4l2m2m
+  # - --filesystem=home:ro
+  # - --filesystem=host:ro
+  # - --filesystem=xdg-download:ro
+  # - --filesystem=xdg-music:ro
+  # - --filesystem=xdg-pictures:ro
+  # - --filesystem=xdg-videos:ro
+
+modules:
```
Comment (orowith2os) [inline org.jellyfin.JellyfinServer.yml:39]: I'd recommend setting up x-checker-data for the sources. It won't update your nuget-generated-sources yet, but it'll handle the heavy stuff for you.

**Comment by istori1** [comment]
> > I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub
> 
> Could you elaborate on your connection to the upstream team, or where the discussion has taken place?

@thornbill can you validate my statements above?


> > They where not interested in building a flatpak, but welcomed the idea of having it created for them as a comunity supported project. I am also an active participant on their developmet rooms.
> 
> Have you done any testing to ensure this works properly compared to, say, a docker container? It could also be useful to upstream some of these files, like the .desktop and .metainfo.xml files.

Also, what is your stance on adding some of the files here to upstream?

**Comment by istori1** [comment]
> > They where not interested in building a flatpak, but welcomed the idea of having it created for them as a comunity supported project. I am also an active participant on their developmet rooms.
> 
> Have you done any testing to ensure this works properly compared to, say, a docker container? It could also be useful to upstream some of these files, like the .desktop and .metainfo.xml files.

In my opinion it works better than a docker container, because it is much easier for an inexperienced user to use it. In terms of functionally it should support everything the docker container does (unless I missed something).

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

File: org.jellyfin.JellyfinServer.yml (hunk 10 above)
Comment (istori1) [reply org.jellyfin.JellyfinServer.yml:39]: Added x-checker-data for the sources.

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
> Is it marked as draft on purpose?

Yes. OpenCL is not working for Intel. Appears to be a missing driver.

**Comment by orowith2os** [comment]
> > Is it marked as draft on purpose?
> Yes. OpenCL is not working for Intel. Appears to be a missing driver. https://github.com/intel/compute-runtime

I'm looking into working with AMD to flatpak ROCm, as well as add a dedicated compute driver mount point for freedesktop-sdk. Ideally the Mesa shipped in freedesktop-sdk would enable RustiCL too. Imo it's not too much of a thing to worry about here.

https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1181
https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1535
https://github.com/RadeonOpenCompute/ROCm/issues/1685
https://github.com/orowith2os/rocm-flatpak

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Comment by istori1** [comment]
bot, build org.jellyfin.JellyfinServer

**Pull request closed without merging**